src/main/java/com/amazon/ion/impl/macro/EncodingContext.kt

```diff
@@ -1,12 +1,15 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
 package com.amazon.ion.impl.macro
 
 /**
  * When we implement modules, this will likely need to be replaced.
  * For now, it is a placeholder for what is to come and a container for the macro table.
  */
-class EncodingContext(
-    val macroTable: Map<MacroRef, Macro>
-) {
+class EncodingContext(macroTable: Map<MacroRef, Macro>) {
+
+    val macroTable = macroTable.toMutableMap()
```

Comment:
This performs a copy, allowing the caller to mutate the map passed in as an argument without modifying the underlying encoding context. This map is mutable because users may add new macros to the table.